src/views/1_ApplicationComponent.tsx

```diff
@@ -133,37 +134,54 @@ export class ApplicationComponent extends React.Component<{}, {}> {
             );
         }
 
-        let sessions = _.flatten(this.activeTab.sessions.map((session, index) => {
-                const isActive = session === this.activeTab.activeSession();
-
-                const sessionComponent = (
-                    <SessionComponent session={session}
-                                      key={session.id}
-                                      style={css.session(isActive, index)}
-                                      isActive={isActive}
-                                      updateStatusBar={isActive ? () => this.forceUpdate() : undefined}
-                                      activate={() => {
-                                   this.activeTab.activateSession(session);
-                                   this.setState({ sessions: this.activeTab.sessions });
-                              }}>
-                    </SessionComponent>
-                );
-
-                if (isActive) {
-                    return [sessionComponent];
-                } else {
-                    return [
-                        sessionComponent,
-                        <div className="shutter" style={css.sessionShutter(index)}></div>,
-                    ];
-                }
+        let renderSessionContainer: any;
```

Comment:
So, all you need to do in order to render them is to map the tree into a grid of areas. Like
```
Horizontal
  Leaf
  Leaf
  Vertical
    Leaf
    Leaf
```
would become

```
'session1 session1'
'session2 session2'
'session3 session4'
```